log.go

```diff
@@ -140,3 +131,14 @@ type LeveledLoggerInterface interface {
 	// Warnf logs a warning message using Printf conventions.
 	Warnf(format string, v ...interface{})
 }
+
+//
+// Private functions
+//
+
+func (l *LeveledLogger) printLog(format string, tag string, level Level, output io.Writer, v ...interface{}) {
+	if l.Level >= level {
+		s := fmt.Sprintf(format, v...)
+		fmt.Fprintf(output, "[%s] %s\n", tag, s)
```

Comment:
What do you think about collapsing these two print calls into just one. i.e.

``` go
fmt.Printf(output, "[%s] " + format + "\n", tag v...)
```

(Not sure that exact line works, but something like it should.)

I gotta say too that I'm generally a fan of extracting things into small functions, but I'm not sure that we're getting our money's worth here. The previous standard of 3 lines per error/warn/info was really not that bad, and we've made this incrementally more difficult to understand in that `if l.Level >= level` is harder to read than `if l.Level >= LevelWarn {`.